modules/seqexec/server/src/main/scala/seqexec/server/gems/Gems.scala

```diff
@@ -24,25 +33,71 @@ trait Gems[F[_]] extends Gaos[F] {
 
 object Gems {
 
-  private class GemsImpl[F[_]: Applicative] (controller: GemsController[F], config: GemsConfig) extends Gems[F] {
+  private class GemsImpl[F[_]: MonadError[?[_], Throwable]] (controller: GemsController[F],
+                                                             config: GemsConfig,
+                                                             guideConfigDb: GuideConfigDb[F]
+                                                            ) extends Gems[F] {
 
     override val cfg: GemsConfig = config
 
-    override def observe(config: Either[AltairConfig, GemsConfig], expTime: Time): F[Unit] = controller.observe(expTime)
+    override def observe(config: Either[AltairConfig, GemsConfig], expTime: Time): F[Unit] = ().pure[F]
 
-    override def endObserve(config: Either[AltairConfig, GemsConfig]): F[Unit] = controller.endObserve
+    override def endObserve(config: Either[AltairConfig, GemsConfig]): F[Unit] = ().pure[F]
 
-    override def pauseResume(config: GemsConfig,
-                             pauseReasons: PauseConditionSet,
-                             resumeReasons: ResumeConditionSet): F[PauseResume[F]] = {
-      controller.pauseResume(cfg, pauseReasons, resumeReasons)
-    }
+    override def pauseResume(pauseReasons: PauseConditionSet,
+                             resumeReasons: ResumeConditionSet): F[PauseResume[F]] =
+      guideConfigDb.value.map(_.gaosGuide).flatMap{
+        case Some(Right(gemsCfg)) => controller.pauseResume(pauseReasons, resumeReasons)(combine(gemsCfg, cfg))
+        case _                    =>
+          SeqexecFailure.Execution("Attempting to run GeMS sequence before GeMS has being configured.")
+            .raiseError[F, PauseResume[F]]
+      }
 
     override val stateGetter: GemsWfsState[F] = controller.stateGetter
   }
 
-  def fromConfig[F[_]: Applicative](controller: GemsController[F])/*(config: Config)*/: F[Gems[F]] =
-    Applicative[F].pure(new GemsImpl[F](controller, GemsOff))
+  // `combine` calculates the final configuration between the configuration coming from the step and the configuration
+  // set by the operator.
+  private def combine(opConfig: GemsConfig, stepConfig: GemsConfig): GemsConfig = GemsController.GemsOn(
```

Comment:
Could this be a `Monoid`? then you could do `config1 |+| config2`

Reply:
No, because it is not commutative. The configuration set by the operator in TCC takes precedence.

Reply:
Monoids only have to be associative, not commutative. `this orElse that` is a common pattern.